cnf-certification-test/operator/suite.go

```diff
@@ -17,14 +17,47 @@
 package operator
 
 import (
+	"fmt"
+
 	"github.com/onsi/ginkgo/v2"
-	"github.com/sirupsen/logrus"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
```

Comment:
to have an idea about the versioning 
please take a look here
https://kubernetes.io/docs/reference/using-api/deprecation-policy/
in short, here's the priority (and we should avoid alpha API)
*v1
*v1beta2
*v1beta1
*v1alphaX

Reply:
I'm not sure how to proceed, then. This is the list of available api versions for the operator-framework api clienset:
```
type Clientset struct {
	*discovery.DiscoveryClient
	operatorsV1alpha1 *operatorsv1alpha1.OperatorsV1alpha1Client
	operatorsV1alpha2 *operatorsv1alpha2.OperatorsV1alpha2Client
	operatorsV1       *operatorsv1.OperatorsV1Client
	operatorsV2       *operatorsv2.OperatorsV2Client
}
```
I chose operatorsV1alpha1 for two reasons:
1. That's the api that "oc get csv" seems to be using (tested with -v=6)
```
greyerof@polimnia ~/github/tnf-new-arch (operator_tc_migration1)$ oc get csv -A -v=6
I0217 12:03:35.912175 3178403 loader.go:379] Config loaded from file:  /home/greyerof/.kube/config
I0217 12:03:35.937802 3178403 round_trippers.go:445] GET https://192.168.49.2:8443/apis/operators.coreos.com/v1alpha1/clusterserviceversions?limit=500 200 OK in 19 milliseconds
NAMESPACE         NAME                    DISPLAY          VERSION   REPLACES   PHASE
default           nginx-operator.v0.0.1   test-operator    0.0.1                Succeeded
kube-node-lease   nginx-operator.v0.0.1   test-operator    0.0.1                Succeeded
kube-public       nginx-operator.v0.0.1   test-operator    0.0.1                Succeeded
kube-system       nginx-operator.v0.0.1   test-operator    0.0.1                Succeeded
olm               nginx-operator.v0.0.1   test-operator    0.0.1                Succeeded
olm               packageserver           Package Server   0.18.3               Succeeded
operators         nginx-operator.v0.0.1   test-operator    0.0.1                Succeeded
tnf               nginx-operator.v0.0.1   test-operator    0.0.1                Succeeded
greyerof@polimnia ~/github/tnf-new-arch (operator_tc_migration1)$
```

2. That's the only api version that has an interface that allows you to List (typed) ClusterServiceVersion resources (interface ClusterServiceVersionsGetter).

My first try was using v2, but failed, as that api has only one getter:
```
type OperatorsV2Interface interface {
	RESTClient() rest.Interface
	OperatorConditionsGetter
}
```
Even in the case that we use the v2's RestClient() to make the manual query (Get()), the output needs to be serialized into a ClusterServiceVersion typed struct, which is currently defined in v1alpha1 only, so the dependency with v1alpha1 is unavoidable, I'm afraid, unless I'm missing something else.